packages/mui-utils/src/resolveProps.ts

```diff
@@ -11,10 +20,14 @@ export default function resolveProps<
     slots?: Record<string, unknown>;
     slotProps?: Record<string, unknown>;
   } & Record<string, unknown>,
->(defaultProps: T, props: T) {
+>(defaultProps: T, props: T, mergePropsWhiteList?: Array<string>) {
```

Comment:
How about `shallowMergePropNames` to be more explicit? 
```suggestion
>(defaultProps: T, props: T, shallowMergePropNames?: Array<string> = []) {
```